### PR TITLE
Chore/small fixes

### PR DESCRIPTION
**Pull request by selfuryon**
Small changes:
* rename root `example` folder to `research`
* move docker-compose with dirks to `research` folder
* some nix fixes and checks
* small changes in github actions

### CODE REVIEW

File: research/research.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// nolint
+package main
```
Comment (SpontaneousOverthrow) [inline research/research.go:2]: Update README please, add this folder instead of examples

**Review by SpontaneousOverthrow — CHANGES_REQUESTED** [review]
Update read

File: research/research.go (hunk 1 above)
Comment (SpontaneousOverthrow) [reply research/research.go:2]: `dkc uses herumi/bls-eth-go-binary. The pseudocode of using herumi/bls-eth-go-binary in dkc is available here`

Comment (selfuryon) [reply research/research.go:2]: Done!

**Review by SpontaneousOverthrow — APPROVED** [review]

**Pull request merged**